### PR TITLE
Add RetrofitCallAdapterFactory

### PR DESCRIPTION
**Pull request by bychkovdmitry**
### Description

Introduces RetrofitCallAdapterFactory which allows us to use chat Call as a return type in RetrofitApi

### Checklist

- [X] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [X] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes
- [X] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [X] Reviewers added

### CODE REVIEW

**Review by JcMinarro — COMMENTED** [review]
Can't we totally remove `RetrofitCallMapper` class?
On the other hand, `ChatApi` interface seems useless, it is only implemented by `ChatApiImpl` What do you think about to move `ChatApiImpl` code to `ChatApi` and convert it to class?

**Comment by bychkovdmitry** [comment]
> Can't we totally remove `RetrofitCallMapper` class?
> On the other hand, `ChatApi` interface seems useless, it is only implemented by `ChatApiImpl` What do you think about to move `ChatApiImpl` code to `ChatApi` and convert it to class?

Removed ChatApi interface, removed RetrofitCallMapper. Added possibility to get the original retrofit call from chat call for this case:

`retrofitCdnApi .sendImage(...).originalCall().enqueue(RetroProgressCallback(callback))`

File: client/src/main/java/io/getstream/chat/android/client/call/RetrofitCall.kt [hunk 1]
```diff
@@ -35,6 +35,8 @@ internal class RetrofitCall<T : Any>(
         call.cancel()
     }
 
+    fun originalCall(): retrofit2.Call<T> = call
```
Comment (zsmb13) [inline client/src/main/java/io/getstream/chat/android/client/call/RetrofitCall.kt]: Nit: could be exposed as just the property itself, not sure we need this separate getter for it.

File: client/src/main/java/io/getstream/chat/android/client/api/RetrofitCallAdapterFactory.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            throw IllegalArgumentException("Call return type must be parameterized as Call<Foo>")
+        }
+        val responseType: Type = getParameterUpperBound(0, returnType)
+        return RetrofitCallAdapter<Any>(responseType, chatParser)
+    }
+
+    companion object {
+        @JvmStatic
+        fun create(chatParser: ChatParser) = RetrofitCallAdapterFactory(chatParser)
+    }
```
Comment (zsmb13) [inline client/src/main/java/io/getstream/chat/android/client/api/RetrofitCallAdapterFactory.kt]: Nit: we don't need `@JvmStatic`, as we don't have Java code 😄 . We could also drop this factory method, and just use the constructor of the class directly.

File: client/src/test/java/io/getstream/chat/android/client/api/RetrofitCallAdapterFactoryTests.kt [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import okhttp3.mockwebserver.MockWebServer
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import retrofit2.CallAdapter
+import retrofit2.Retrofit
+import java.lang.reflect.Type
+
+internal class RetrofitCallAdapterFactoryTests {
```
Comment (zsmb13) [inline client/src/test/java/io/getstream/chat/android/client/api/RetrofitCallAdapterFactoryTests.kt:16]: Nice coverage of the different cases here 👍

**Review by zsmb13 — DISMISSED** [review]
Looks great, I have just a couple small comments

**Review by zsmb13 — APPROVED** [review]

**Pull request merged**